pkg/webui/components/key-value-map/entry.js

```diff
@@ -27,117 +26,102 @@ const m = defineMessages({
   deleteEntry: 'Delete entry',
 })
 
-class Entry extends React.Component {
-  static propTypes = {
-    readOnly: PropTypes.bool,
-  }
-  static defaultProps = { readOnly: false }
-
-  _getKeyInputName() {
-    const { name, index } = this.props
-
-    return `${name}[${index}].key`
-  }
-
-  _getValueInputName() {
-    const { name, index } = this.props
-
-    return `${name}[${index}].value`
-  }
-
-  @bind
-  handleRemoveButtonClicked(event) {
-    const { onRemoveButtonClick, index } = this.props
-
-    onRemoveButtonClick(index, event)
-  }
-
-  @bind
-  handleKeyChanged(newKey) {
-    const { onChange, index } = this.props
-
-    onChange(index, { key: newKey })
-  }
-
-  @bind
-  handleValueChanged(newValue) {
-    const { onChange, index } = this.props
-
-    onChange(index, { value: newValue })
-  }
-
-  @bind
-  handleBlur(event) {
-    const { name, onBlur, value } = this.props
-
-    const { relatedTarget } = event
-    const nextTarget = relatedTarget || {}
-
-    if (
-      nextTarget.name !== this._getKeyInputName() &&
-      nextTarget.name !== this._getValueInputName()
-    ) {
-      onBlur({
-        target: {
-          name,
-          value,
-        },
-      })
-    }
-  }
-
-  render() {
-    const {
-      keyPlaceholder,
-      valuePlaceholder,
-      value,
-      indexAsKey,
-      readOnly,
-      inputElement: InputElement,
-      additionalInputProps,
-    } = this.props
-
-    return (
-      <div className={style.entriesRow}>
-        {!indexAsKey && (
-          <InputElement
-            data-test-id={this._getKeyInputName()}
-            className={style.input}
-            name={this._getKeyInputName()}
-            placeholder={keyPlaceholder}
-            type="text"
-            onChange={this.handleKeyChanged}
-            onBlur={this.handleBlur}
-            value={value.key}
-            readOnly={readOnly}
-            code
-            {...additionalInputProps}
-          />
-        )}
+const Entry = ({
+  readOnly,
+  name,
+  value,
+  index,
+  onRemoveButtonClick,
+  onChange,
+  onBlur,
+  inputElement: InputElement,
+  indexAsKey,
+  valuePlaceholder,
+  keyPlaceholder,
+  additionalInputProps,
+}) => {
+  const _getKeyInputName = useCallback(() => `${name}[${index}].key`, [index, name])
+
+  const _getValueInputName = useCallback(() => `${name}[${index}].value`, [index, name])
```

Comment:
It's not really purposeful to recreate this with a `useCallback()`. This is a better use case for `useMemo()`.